.github/workflows/release-build.yml

```diff
@@ -0,0 +1,85 @@
+name: "Release Build"
+
+on:
+  workflow_dispatch:
+  release:
+    types: [published]
+
+jobs:
+  build:
+    runs-on: windows-2022
+    strategy:
+      matrix:
+        Game: 
+          - Ares
+          - TS
+          - YR
+    
+    outputs:
+      GitVersion_SemVer: ${{ steps.set_gitversion_outputs.outputs.GitVersion_SemVer }}
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Setup .NET SDK
+      uses: actions/setup-dotnet@v4
+      with:
+        global-json-file: ./global.json
+
+    - name: Build ${{matrix.Game}}
+      run: ./Scripts/build.ps1 ${{matrix.Game}}
+      shell: pwsh
+
+    - name: Development Build Check
+      shell: pwsh
+      run: |
+        Write-Output "GitVersion_BranchName: $env:GitVersion_BranchName"
+        Write-Output "GitVersion_CommitsSinceVersionSource: $env:GitVersion_CommitsSinceVersionSource"
+        if ($env:GitVersion_CommitsSinceVersionSource -ne "0") {
+          Write-Output "::error:: This is a development build and should not be released. Do you forget creating a new tag for the release?"
+          exit 1
+        }
+
+    - uses: actions/upload-artifact@v4
+      name: Upload Artifacts
+      with:
+        name: artifacts-${{matrix.Game}}
+        path: ./Compiled/${{matrix.Game}}
+
+    - name: Set GitVersion Outputs
+      id: set_gitversion_outputs
+      shell: pwsh
+      run: |
+        echo "GitVersion_SemVer=$env:GitVersion_SemVer" >> $env:GITHUB_OUTPUT
+
+  package:
+    needs: build
+    runs-on: windows-2022
+    env:
+      ARCHIVE_NAME: ${{ format('xna-cncnet-client-artifacts-{0}.7z', needs.build.outputs.GitVersion_SemVer) }}
+
+    steps:
+    - name: Download Build Outputs
+      uses: actions/download-artifact@v4
+      with:
+        path: ./artifacts/
```

Comment:
I am not sure why would you need 2 separate jobs?

Reply:
```yml
    strategy:
      matrix:
        Game: 
          - Ares
          - TS
          - YR
```

This repeats the whole steps in the job for each game, while we need to upload the final artifact for only once.

Reply:
I don't agree we need that. We can and should upload 3 separate archives. They will upload to the same release. Please see how I've done it in the spawner workflow:

![image](https://github.com/user-attachments/assets/55751302-2e1d-4334-8d51-0238fb489d3f)

https://github.com/CnCNet/yrpp-spawner/blob/v0.0.0.7/.github/workflows/release-build.yml

Reply:
> I don't agree we need that. We can and should upload 3 separate archives. They will upload to the same release. Please see how I've done it in the spawner workflow:
> 
> 
> 
> ![image](https://github.com/user-attachments/assets/55751302-2e1d-4334-8d51-0238fb489d3f)
> 
> 
> 
> https://github.com/CnCNet/yrpp-spawner/blob/v0.0.0.7/.github/workflows/release-build.yml

The 3 games shares so many common files that it is beneficial to distributing them as a single archive, for a nearly 3 times compression.

Reply:
Beneficial for who? People usually don't need all of them at once, usually a developer only develops one engine mod at a time.

Reply:
> Beneficial for who? People usually don't need all of them at once, usually a developer only develops one engine mod at a time.

*Beneficial Situations*

For example, China has blocked GitHub with random interference. Sometimes a user can open GitHub, sometimes one can't. Personally, I can download GitHub Release assets in a limited bandwidth via a proxy, with speed ~100KB/s. Some people might have to ask for another guy help him download the file, and in this situation, merging the three into one increase the chance of 'cache hit'.

*Almost Free Merging*

artifacts-YR.7z -- 12.7 MB
artifacts-TS.7z -- 12.7 MB
artifacts-Ares.7z -- 12.7 MB
all.7z -- 14.1 MB

Only little overhead are added to a person who only seek for one engine. Worth a trade.